### ISSUE TITLE
MySQL: models.XTable.UpsertMany not working

### ISSUE THREAD

**Issue by jacobmolby** [stephenafamo/bob#56]
The `UpsertMany` method on a table for MySQL produces wrong SQL.

Example DDL:
```sql
CREATE TABLE `test` (
  `id` int NOT NULL AUTO_INCREMENT,
  `reference` varchar(255) ,
  `num` int,
  PRIMARY KEY (`id`),
  UNIQUE KEY `reference` (`reference`) USING BTREE
)
```
Given the following table contents:
| id | reference | num |
|--------|--------|--------|
| 1 | a | 1 |
| 2 | b | 1 | 

And this go code:

```go
models.TestsTable.UpsertMany(
	context.Background(),
	bob.Debug(db),
	true,
	[]string{
		models.ColumnNames.Tests.Num,
	},
	&models.TestSetter{
		Num:       omit.From(2),
		Reference: omit.From("a"),
	},
	&models.TestSetter{
		Num:       omit.From(2),
		Reference: omit.From("b"),
	},
)
``` 

It produces the following SQL:

```sql
INSERT INTO `test`  (`reference`, `num`)
VALUES (?, ?), (?, ?)
ON DUPLICATE KEY UPDATE
`num` = `test`.`num`

0: omit.Val[string]{value:"a", state:1}
1: omit.Val[int]{value:2, state:1}
2: omit.Val[string]{value:"b", state:1}
3: omit.Val[int]{value:2, state:1}
```

Which doesn't update the `num` field to the new values.

There are at least two solutions.
1. Use an alias

If using an alias, the resulting SQL should look something similar to this:

```sql
INSERT INTO `test`  (`reference`, `num`)
VALUES ("a", 2), ("b", 2) AS `new`
ON DUPLICATE KEY UPDATE
`num` = `new`.`num`
```

2. Use `VALUES()`

If using an `VALUES()`, the resulting SQL should look something similar to this:

```sql
INSERT INTO `test`  (`reference`, `num`)
VALUES ("a", 2), ("b", 2)
ON DUPLICATE KEY UPDATE
`num` = VALUES(`num`)
```

Both results in the correct behaviour, which the current method does not. 

It could be that I'm completely missing something. If that's the case please let me know 😃

**Comment by stephenafamo** [comment]
This looks like a bug. I'll take a look over the weekend.

**Comment by stephenafamo** [comment]
Try 788ec56

**Comment by jacobmolby** [comment]
@stephenafamo 
It seems to be working, nice job! 🔥

**Comment by stephenafamo** [comment]
Released in `v0.21.1`
